build/Dockerfile

```diff
@@ -1,6 +1,38 @@
-ARG GO_VERSION=1.22
+# Use Ubuntu 16.04 as the base image
+FROM ubuntu:16.04
 
-FROM golang:$GO_VERSION
+# Define Go version
+ARG GO_VERSION=1.22.8
```

Comment:
would be possible to avoid having multiple declarations of the go version, it could be prone to error whenever we need to bump it.

Reply:
This version is only the default version to be used in case we don't get a version from Makefile. But https://github.com/newrelic/nri-flex/blob/5c2e540fc3223c786e9200390b92e0b2b14b9965/build/ci.mk#L7 specifies a build arg which comes from the Makefile.